cluster-autoscaler/core/scale_down.go

```diff
@@ -692,7 +692,10 @@ func (sd *ScaleDown) TryToScaleDown(allNodes []*apiv1.Node, pods []*apiv1.Pod, p
 	scaleDownStatus := &status.ScaleDownStatus{NodeDeleteResults: sd.nodeDeletionTracker.GetAndClearNodeDeleteResults()}
 	nodeDeletionDuration := time.Duration(0)
 	findNodesToRemoveDuration := time.Duration(0)
-	defer updateScaleDownMetrics(time.Now(), &findNodesToRemoveDuration, &nodeDeletionDuration)
+	scaleDownFnStartTime := time.Now()
+	metrics.UpdateLastTime(metrics.ScaleDown, scaleDownFnStartTime)
+	defer updateScaleDownMetrics(scaleDownFnStartTime, &findNodesToRemoveDuration, &nodeDeletionDuration)
```

Comment:
It seems that now we can be calling `updateScaleDownMetrics` twice per call to `TryToScaleDown`.

Reply:
good catch, that's a problem. 

Apart from updating the metrics at every return path, is there anything we can go to have an accurate `metrics.ScaleDownNodeDeletion`?